Code/Framework/AzFramework/AzFramework/Spawnable/Script/SpawnableScriptAssetRef.h

```diff
@@ -36,6 +36,16 @@ namespace AzFramework::Scripts
         AZ::Data::Asset<Spawnable> GetAsset() const;
 
     private:
+        class SerializationEvents : public AZ::SerializeContext::IEventHandler
+        {
+            void OnReadEnd(void* classPtr) override
```

Comment:
After reviewing further, it occurred to me that a spawnable is a product asset. So, if it got updated on disk, is OnAssetReloaded() not called? Why do we need to plug into this SerializationEvent system?

Reply:
yes exactly, that is the issue. OnAssetReloaded() does not get called because the asset gets unloaded by the AssetSystem. I believe the reason this happens is because we are caching entities that were spawned in the past with `spawnedEntities.emplace_back(CloneSingleEntity(*entitiesToSpawn[i], ticket.m_entityIdReferenceMap, *request.m_serializeContext));` - SpawnableScriptAssetRef gets cloned as well, but it never runs the code to connect to AssetBus listener in the copy. So when asset gets eventually refreshed, this copy still holds reference to the old asset instance.

Why the SerializationEvents needed? This is the only place that I can confidently capture the location where m_asset field is modified, update it and connect AssetBus listeener.

Reply:
It's strange you're seeing a problem there. If you're connected on the AssetBus to an asset ID, you should always get a message about that asset if the asset processor re-processes it.

One suspicious thing to me here is the use of AZ::Data::AssetBus::Handler::BusConnect instead of Data::AssetBus::MultiHandler::BusConnect

Reply:
when the entities get cloned, they don't actually call the connection code

Reply:
So, it looks like CloneObjectAndGenerateNewIdsAndFixRefs() may clone the asset data but doesn't trigger onAsset...() notifications. @AMZN-koppersr , wondering if you have seen this before?